0000-frontend-development-infrastructure-and-workflow-changes.md

```diff
@@ -0,0 +1,75 @@
+- Feature Name: frontend-development-infrastructure-and-workflow-changes
+- Start Date: 2016-07-20
+- RFC PR: 
+- Issue: 
+
+# Summary
+[summary]: #summary
+
+Currently, client side javascript in foreman is implemented with native javascript and the jquery library and managed by the Rails asset pipeline. In addition, various unrelated libraries are used for specific tasks, charting for example. This RFC proposes adding the React javascript library to the technology stack via NPM and Webpack. [(see Webpack PR)](https://github.com/theforeman/foreman/pull/3433) 
```

Comment:
I'll also add that its really hard to use 3rd party react components without a proper package manager such as npm.